zerver/views/messages.py

```diff
@@ -971,7 +971,7 @@ def send_message_backend(request, user_profile,
             return json_error(_("User not authorized for this query"))
         (valid_input, mirror_sender) = \
             create_mirrored_message_users(request, user_profile, message_to)
-        if not valid_input:
+        if not valid_input or mirror_sender is None:  # FIXME: Amend to use exceptions?
```

Comment:
Yeah, having `create_mirrored_message_users` raise exceptions would be a reasonable way to clean this up. The main downside is that we'd have to either repeat the "Invalid mirrored message" string at each of the error paths there, or have them raise one exception and then this code catch that to raise another, the `json_error`.

An alternate way would be to have `create_mirrored_message_users` return just one value, an `Optional[UserProfile]`. As it is, we have `mirror_sender` non-None just if `valid_input` is True anyway, so we could cut the redundancy.

Reply:
Yeah, I like that approach.